### PR TITLE
Export use confirm hook and confirmation dialog

### PR DESCRIPTION
**Pull request by VanAnderson**
Export ConfirmationDialog and useConfirm from index.

https://github.slack.com/archives/C01L618AEP9/p1619730448377100

### CODE REVIEW

**Comment by smockle** [comment]
Running `yarn run dist` locally gave me these warnings, which may be related to why CI is failing:

```
[dist:rollup] (!) Circular dependencies
[dist:rollup] src/index.ts -> src/Dialog/ConfirmationDialog.tsx -> src/index.ts
[dist:rollup] src/index.ts -> src/Dialog/ConfirmationDialog.tsx -> src/Dialog/Dialog.tsx -> src/index.ts
[dist:rollup] src/index.ts -> src/ActionList/index.ts -> src/ActionList/List.tsx -> src/ActionList/Item.tsx -> src/index.ts
```

**Review by smockle — APPROVED** [review]

File: src/ActionList/Item.tsx [hunk 1]
```diff
@@ -75,9 +74,9 @@ export interface ItemProps extends React.ComponentPropsWithoutRef<'div'>, SxProp
 const getItemVariant = (variant = 'default', disabled?: boolean) => {
   if (disabled) {
     return {
-      color: themeGet('colors.text.disabled'),
-      iconColor: themeGet('colors.text.disabled'),
-      annotationColor: themeGet('colors.text.disabled'),
+      color: get('colors.text.disabled'),
```
Comment (dgreif) [inline src/ActionList/Item.tsx:77]: Is this the right `get` to use? I think we need `import {themeGet} from '@styled-system/theme-get'` or `import {themeGet} from '../constants'` in order to get proper theming, but I'm not certain.  @colebemis mentioned this on a PR the other day but I can't remember where.

Comment (smockle) [reply src/ActionList/Item.tsx:77]: `get` calls `themeGet` internally, so theme values should be used:
https://github.com/primer/components/blob/7aeb53fec23a80414f6c1db8b31c06beb2073dd2/src/constants.ts#L8

Comment (dgreif) [reply src/ActionList/Item.tsx:77]: @smockle wouldn't we still need to import `get` here, or is it stored on `global` somewhere?

**Review by dgreif — APPROVED** [review]

**Pull request merged**